### ISSUE TITLE
After pattern (testcase: tmp_after01.cfr)

### ISSUE THREAD

**Issue by RaoMukkamala** [juodaspaulius/BehavioralClafer#1]
The alloy code generated for the testcase failed to compile in alloy 4.2.
Here are the notes:
    Checks
    1.  test case is OK ? Yes
    2. Compiles ? Yes
    3. De-sugars correctly ? Yes
    4. Compiles in Alloy ? No. 

```
Error message:  
Starting the solver...
A syntax error has occurred: 
The name "@r_this" cannot be found.
More info: In line 30, @r_this. is undefined
@r_this should be changed to  this 

(some loop and all t':t.*(Time <: next + loop) | (some @r_this.@r_c0_c.t') => ((some loop and all t'':t'.*(Time <: next + loop) | no @r_this.@r_c0_d.t''))) }
```
1. Instances in alloy are OK? Yes
   After fixing the syntax in alloy, I tried to run many solutions for instances in alloy 4.2 and projected over the time to check the evolution  of  instances. 
   I have not found any contradictory instances, even though I was not able to produce any 'd' instance attached to root through its ancestor.
   So it looks OK.

**Comment by RaoMukkamala** [comment]
After the new release, the generated alloy code does not give any compile errors. 
when I tested the instance generation, one thing bothers me is that even through it is specified the cardinality as ? (0..1) on  b, c, d clafers (also on clafer a), the instance generator generates a lot of instances (e.g. c0_b$0, c0_b$1, c0_b$2, c0_b$3, c0_b$4, c0_b$5, c0_b$6, c0_b$7). Of course, it is not a violation, as at any point only one element is connected to the root. 
I have tried several times, but I failed to generate a valid 'd' clafer attached to root. 
We need to check does this constraint '[never d after c ]'  prevents a valid instance of having 'd' connected to root? (of course before happening of c)
